src/components/scene/vr-mode-ui.js

```diff
@@ -81,12 +57,74 @@ module.exports.Component = registerComponent('vr-mode-ui', {
     });
   },
 
-  hide: function () {
+  attachEventListeners: function () {
+    var self = this;
+    var scene = this.el;
+
+    // Hide/show VR UI when entering/exiting VR mode.
+    scene.addEventListener('enter-vr', this.onEnterVR.bind(this));
+    scene.addEventListener('exit-vr', this.onExitVR.bind(this));
+
+    window.addEventListener('message', function (event) {
+      if (event.data.type === 'loaderReady') {
+        self.insideLoader = true;
+        self.remove();
+      }
+    });
+
+    // Modal that tells the user to change orientation if in portrait.
+    window.addEventListener('orientationchange', this.toggleOrientationModalIfNeeded.bind(this));
+    this.toggleOrientationModalIfNeeded();
+  },
+
+  onEnterVR: function () {
+    this.toggleEnterVRButtonIfNeeded();
+    this.toggleOrientationModalIfNeeded();
+  },
+
+  onExitVR: function () {
+    this.toggleEnterVRButtonIfNeeded();
+    this.hideOrientationModal();
+  },
+
+  hideEnterVRButton: function () {
+    if (!this.enterVREl) { return; }
     this.enterVREl.classList.add(HIDDEN_CLASS);
   },
 
-  show: function () {
+  showEnterVRButton: function () {
+    if (!this.enterVREl) { return; }
     this.enterVREl.classList.remove(HIDDEN_CLASS);
+  },
+
+  toggleEnterVRButtonIfNeeded: function () {
```

Comment:
I usually don't have issues with verbose names. Less space for missinterpretation and autocompletion in your editor helps with the typing